### PR TITLE
Searchbox: use EventListener, not EventGroup.

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Similar to #11949, replacing usage of EventGroup with EventListener in SearchBox because it's more performant and will help with our convergence effort as it's from `fluentui/react-component-event-listener` package. Had to add `onBlur` to the SearchBox.types.ts file and have the type be `(ev?: any) => void;` to make it compatible with the `listener` argument of EventListener - tried to make it not `any` and instead `FocusEvent` but caused an error that said I need to change the whole interface's type (incompatible with `ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElement>`).

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/11960)

### CODE REVIEW

**Comment by aneeshack4** [comment]
Hmm SizeAuditor is throwing a strange webpack error

**Comment by ecraig12345** [comment]
> Hmm SizeAuditor is throwing a strange webpack error

The error is `Error: Cannot find module '@uifabric/experiments'`

I think this is because `yarn buildto test-bundles` isn't building experiments (it's not listed in the log in the previous step). @kenotron  Any idea why this would suddenly happen when she didn't change any deps?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 1]
```diff
@@ -108,35 +101,38 @@ export class SearchBoxBase extends React.Component<ISearchBoxProps, ISearchBoxSt
...
+              <IconButton
+                styles={{ root: { height: 'auto' }, icon: { fontSize: '12px' } }}
+                iconProps={{ iconName: 'Clear' }}
+                {...clearButtonProps}
+                onClick={this._onClearClick}
+              />
+            </div>
+          )}
+        </div>
+        {hasFocus && <EventListener targetRef={this._rootElement} type="blur" capture listener={this._onBlur} />}
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: Should this be inside the div instead?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts [hunk 2]
```diff
@@ -60,6 +60,11 @@ export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElem
    */
   onEscape?: (ev?: any) => void;
 
+  /**
+   * Callback executed when the user moves focus away from the search box.
+   */
+  onBlur?: (ev?: any) => void;
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: Do you know the type of the event?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 3]
```diff
@@ -108,35 +101,38 @@ export class SearchBoxBase extends React.Component<ISearchBoxProps, ISearchBoxSt
...
-        />
-        {value!.length > 0 && (
-          <div className={classNames.clearButton}>
-            <IconButton
-              styles={{ root: { height: 'auto' }, icon: { fontSize: '12px' } }}
-              iconProps={{ iconName: 'Clear' }}
-              {...clearButtonProps}
-              onClick={this._onClearClick}
-            />
+      <>
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: remove fragment, unneeded.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx (hunk 1 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: yup

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts (hunk 2 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: I think you can remove this.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 4]
```diff
@@ -236,10 +227,7 @@ export class SearchBoxBase extends React.Component<ISearchBoxProps, ISearchBoxSt
...
   };
 
-  private _onBlur = (ev: React.FocusEvent<HTMLInputElement>): void => {
-    if (this._events) {
-      this._events.off();
-    }
+  private _onBlur = (ev: FocusEvent): void => {
     this.setState({
       hasFocus: false
     });
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: to fix the red squiggle on `this.props.onBlur(ev)`, the best you can do here (I think) is to cast the `ev` to an `any` to use the `HTMLInputAttributes` onBlur definition. It's not technically perfect, but it's also a non breaking change that doesn't really matter a ton.

**Comment by kenotron** [comment]
> > Hmm SizeAuditor is throwing a strange webpack error
> 
> The error is `Error: Cannot find module '@uifabric/experiments'`
> 
> I think this is because `yarn buildto test-bundles` isn't building experiments (it's not listed in the log in the previous step). @kenotron Any idea why this would suddenly happen when she didn't change any deps?

I think it isn't because of the experiments thing right? I think the build is failing because there's no mention of the new event package in the oufr package.json in this PR.

**Comment by kenotron** [comment]
I'm adding a label for not merging here because the code is not ready to published used until repo merge has happened I believe.

**Comment by ecraig12345** [comment]
LGTM but agreed with Ken that we need to wait on merging.

**Pull request closed without merging**